block/manager.go

```diff
@@ -101,11 +101,22 @@ type Manager struct {
 
 // getInitialState tries to load lastState from Store, and if it's not available it reads GenesisDoc.
 func getInitialState(store store.Store, genesis *cmtypes.GenesisDoc) (types.State, error) {
+	b, err := store.GetBlock(uint64(genesis.InitialHeight))
```

Comment:
since `genesis.InitialHeight` is the height for next block to be produced, it will always hit the `NewFromGenesisDoc`. I think you can do the whole logic like:
```
s, err := store.GetState()
if err != nil {
     // Starting from scratch
     s, err = types.NewFromGenesisDoc(genesis)
}
// Pick-up where we left last time, without overwriting the state
// Make sure the last block height from the loaded state and genesis initial heights are consecutive 
if s.LastBlockHeight + 1 != genesis.InitialHeight {
     return types.State{}, fmt.Errorf("")
}
// Make sure the loaded app hash matches the genesis 
if !bytes.Equal(s.AppHash, genesis.AppHash.Bytes()) {
     return types.State{}, fmt.Errorf("")
}
```

Reply:
Why make sure they're consecutive? `if s.LastBlockHeight + 1 != genesis.InitialHeight` will always throw an error, because your loaded state will usually be newer than the supplied genesis

Reply:
did you mean something like this?
```
// Pick-up where we left last time, without overwriting the state
// Make sure the last block height from the loaded state and genesis initial heights are consecutive 
if s.LastBlockHeight + 1 == genesis.InitialHeight {
  // Make sure the loaded app hash matches the genesis 
  if !bytes.Equal(s.AppHash, genesis.AppHash.Bytes()) {
       return types.State{}, fmt.Errorf("")
  }
}
return s, nil
```

Reply:
i think our assumptions are not same. why is the loaded state newer than the genesis for the hardfork? is that normal?

Reply:
getInitialState is called every time rollkit starts, not just on a fresh hard-fork